src/api/spec/models/user_spec.rb

```diff
@@ -22,11 +22,7 @@
 
     it { expect(user.state).to eq(User::STATES['unconfirmed']) }
 
-    it 'validates uniqueness of login' do
-      invalid_user = build(:user, login: user.login)
-      invalid_user.save
-      expect(invalid_user.errors.full_messages).to eq(['Login is the name of an already existing user.'])
-    end
+    it { is_expected.to validate_uniqueness_of(:login).with_message('is the name of an already existing user.') }
```

Comment:
I got this error now: https://travis-ci.org/bgeuken/open-build-service/jobs/111499790

"
Failure/Error:
       project.relationships.create(user: self, role: Role.find_by_title('maintainer'))

     ActiveRecord::RecordNotSaved:
       You cannot call create unless the parent is saved
"

Might be a race condition. Gone try to get around this